cli/provider/local.go

```diff
@@ -124,37 +120,25 @@ func (p LocalProvider) GetMod(_ context.Context, modReference string) (*ficsit.G
 }
 
 func (p LocalProvider) ModVersionsWithDependencies(_ context.Context, modID string) ([]resolver.ModVersion, error) {
-	cachedModFiles, err := cache.GetCacheMod(modID)
+	modVersions, err := localregistry.GetModVersions(modID)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get cache: %w", err)
+		return nil, fmt.Errorf("failed to get local mod versions: %w", err)
 	}
 
-	versions := make([]resolver.ModVersion, 0)
-
-	for _, modFile := range cachedModFiles {
-		versions = append(versions, resolver.ModVersion{
-			ID:          modID + ":" + modFile.Plugin.SemVersion,
-			Version:     modFile.Plugin.SemVersion,
-			GameVersion: modFile.Plugin.GameVersion,
-		})
-	}
+	// TODO: only list as available the versions that have at least one target cached
```

Comment:
What about this incomplete TODO?

Reply:
Functionally, everything works, but when removing and re-adding a mod while offline, the resolver would pick the latest version, even if it wasn't cached, so the user would have to pick the specific version to use.

Compared to some package managers, pnpm has this same behavior, while dart pub only lists cached versions (very similar to our previous offline implementation), so it would pick the latest cached.

Listing only versions with at least one target cached would reduce the likelihood of the user having to manually choose a version when doing this, but not entirely solve it, for example the case when a client-only profile cached a newer client version of a mod, but a shared profile uses an older version.